### PR TITLE
update Quaternion to use Check

### PR DESCRIPTION
**Pull request by austinEng**
part of #4794

### CODE REVIEW

File: Source/Core/Quaternion.js [hunk 1]
```diff
@@ -1032,12 +923,8 @@ define([
         if (!defined(q0) || !defined(q1) || !defined(s0) || !defined(s1)) {
             throw new DeveloperError('q0, q1, s0, and s1 are required.');
         }
```
Comment (lilleyse) [inline Source/Core/Quaternion.js]: Also check these ones.

Comment (lilleyse) [inline Source/Core/Quaternion.js]: Same with `computeInnerQuadrangle` and `squad`.

Comment (lilleyse) [inline Source/Core/Quaternion.js]: Technically it adds more lines of code - but its good to be consistent.

Comment (austinEng) [inline Source/Core/Quaternion.js]: Updated

**Comment by lilleyse** [comment]
Looks good!

**Pull request merged**